homeassistant/components/recollect_waste/__init__.py

```diff
@@ -69,3 +70,32 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate an old config entry."""
+    version = entry.version
+
+    LOGGER.debug("Migrating from version %s", version)
+
+    # 1 -> 2: Update unique ID of existing, single sensor entity to be consistent with
+    # common format for platforms going forward:
+    if version == 1:
+        version = entry.version = 2
+
+        @callback
+        def migrate_unique_id(entity_entry: er.RegistryEntry) -> dict[str, Any]:
+            """Migrate the unique ID to a new format."""
+            return {
+                "new_unique_id": (
+                    f"{entry.data[CONF_PLACE_ID]}_"
+                    f"{entry.data[CONF_SERVICE_ID]}_"
+                    "current_pickup"
+                )
+            }
+
+        await er.async_migrate_entries(hass, entry.entry_id, migrate_unique_id)
```

Comment:
Should we call `hass.config_entries.async_update_entry(entry)` ?

Reply:
I guess not actually since if it returned True

```
            if result:
                # pylint: disable=protected-access
                hass.config_entries._async_schedule_save()
```

Its just a bit weird that .version is mutable but .data is not